apis/cassandra/v1beta1/cassandradatacenter_types.go

```diff
@@ -394,6 +394,10 @@ type CassandraDatacenterStatus struct {
 
 	// +optional
 	ObservedGeneration int64 `json:"observedGeneration,omitempty"`
+
+	// TrackedTasks tracks the tasks for completion that were created by the cass-operator
+	// +optional
+	TrackedTasks []corev1.ObjectReference `json:"trackedTasks,omitempty"`
```

Comment:
Status updates should not trigger a reconcile. That's a bug in the k8ssandra-operator if that happens and means the watcher is missing version check.

Reply:
TBH I forgot about GenerationChangedPredicate when I wrote my comment. Status updates won't trigger a reconcile for cass-operator but it will for k8ssandra-operator since the latter does not use a GenerationChangedPredicate for its CassandraDatacenter watch. This is by design though. k8ssandra-operator doesn't use GenerationChangedPredicate since it does not manage the CassandraDatacenter. We do want status updates to trigger reconciliation in k8ssandra-operator. Aside from requeues (or lack thereof) I would still be in favor removing TrackedTasks since the information is already available through the CassandraTask.

Reply:
That information isn't actually available, since TrackedTasks only include tasks generated by the cass-operator, while fetching with labels would give back all the tasks generated by anyone. At that point, I would need to know which ones have been created before the operation I ran in the cass-operator and which later. I don't have that information at the moment.